source/DSCResources/DSC_CMCollections/DSC_CMCollections.psm1

```diff
@@ -0,0 +1,685 @@
+
+        [Parameter()]
+        [String[]]
+        $ExcludeMembership,
+
+        [Parameter()]
+        [String[]]
+        $DirectMembership,
+
+        [Parameter()]
+        [ValidateSet('Present','Absent')]
+        [String]
+        $Ensure = 'Present'
+    )
+
+    Import-ConfigMgrPowerShellModule
+    Set-Location -Path "$($SiteCode):\"
+
+    try
+    {
+        $state = Get-TargetResource -SiteCode $SiteCode -CollectionName $CollectionName -CollectionType $CollectionType
+        $cn = Get-CMCollection -Name $CollectionName
+
+        if ($Ensure -eq 'Present')
+        {
+            if ($null -eq $cn)
+            {
+                Write-Verbose -Message ($script:localizedData.CollectionAbsent -f $CollectionName)
+
+                $newCollection = @{
+                    Name                   = $CollectionName
+                    CollectionType         = $CollectionType
+                    LimitingCollectionName = $LimitingCollectionName
+                }
+
+                New-CMCollection @newCollection
+            }
+
+            $buildingParams = @{
+                Name = $CollectionName
+            }
+
+            $itemsForEval = @(
+                @{
+                    Name  = 'Comment'
+                    Value = $Comment
+                }
+                @{
+                    Name  = 'RefreshType'
+                    Value = $RefreshType
+                }
+            )
+
+            foreach ($item in $itemsForEval)
+            {
+                if ((-not [string]::IsNullOrEmpty($item.Value)) -and ($state[$item.Name] -ne $item.Value))
+                {
+                    Write-Verbose -Message ($script:localizedData.CollectionSetting -f $CollectionName, `
+                        $($item.name), $($item.Value), $($state[$item.Name]))
+
+                    $buildingParams += @{
+                        $($item.Name) = $($item.Value)
+                    }
+                }
+            }
+
+            if (-not [string]::IsNullOrEmpty($RefreshSchedule))
+            {
+                $newSchedule = @{
+                    RecurInterval = $RefreshSchedule.RecurInterval
+                    RecurCount    = $RefreshSchedule.RecurCount
+                }
+                
+                $desiredRefreshSchedule = New-CMSchedule @newSchedule
+
+                if ($state.RefreshSchedule)
+                {
+                    $array = @('DayDuration','DaySpan','HourDuration','HourSpan','IsGMT','MinuteDuration','MinuteSpan')
+
+                    foreach ($item in $array)
+                    {
+                        if (($desiredRefreshSchedule).$($item) -ne ($state.RefreshSchedule).$($item))
+                        {
+                            $setSchedule = $true
+                        }
+                    }
+                }
+                else
+                {
+                    $setSchedule = $true
+                }
+            }
+
+            if ($setSchedule)
```

Comment:
You could move this code in the `if ($state.RefreshSchedule)` block and then get rid of the else.Wh

Reply:
if state.RefreshSchedule is null and a schedule was specified would still want to specify to set a schedule

Reply:
Yeah. It could be replaced in both sections. It is doign another logic decision when isn't really needed.